pyro/infer/tracegraph_elbo.py

```diff
@@ -219,37 +219,49 @@ def loss_and_grads(self, model, guide, *args, **kwargs):
         Performs backward on the latter. Num_particle many samples are used to form the estimators.
         If baselines are present, a baseline loss is also constructed and differentiated.
         """
+
+        loss, surrogate_loss = self._loss_and_surrogate_loss(model, guide, *args, **kwargs)
+
+        torch_backward(surrogate_loss, retain_graph=self.retain_graph)
+
+        loss = torch_item(loss)
+        warn_if_nan(loss, "loss")
+        return loss
+
+    def _loss_and_surrogate_loss(self, model, guide, *args, **kwargs):
+
         loss = 0.0
-        weight = 1./self.num_particles
+        surrogate_loss = 0.0
+
         for model_trace, guide_trace in self._get_traces(model, guide, *args, **kwargs):
-            loss += self._loss_and_grads_particle(weight, model_trace, guide_trace)
-        return loss
 
-    def _loss_and_grads_particle(self, weight, model_trace, guide_trace):
+            lp, slp = self._loss_and_surrogate_loss_particle(model_trace, guide_trace, *args, **kwargs)
+            loss += lp
+            surrogate_loss += slp
+
+        loss /= self.num_particles
+        surrogate_loss /= self.num_particles
```

Comment:
Can you avoid in-place operators, since this is now being used by `JitTraceGraph_ELBO`?